### PR TITLE
Restore portable schema changes, with simple LogicalType support

### PR DESCRIPTION
**Pull request by TheNeuralBit**
Un-reverts changes to the portable schema representation, with a couple of changes (https://github.com/apache/beam/commit/22f38527799f0863d74ee3d1e92011be2fd72c28):
- Changes the LogicalType string args to a byte payload
- Supports encoding arbitrary Java SDK LogicalTypes with the `beam:fieldtype:javasdk` URN. Long-term I would like to move types to a registry and discourage this URN ([BEAM-7855](https://issues.apache.org/jira/browse/BEAM-7855)), but for now it replicates the current implementation.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by TheNeuralBit** [comment]
Run Dataflow ValidatesRunner

**Comment by TheNeuralBit** [comment]
Run JavaPortabilityApi PreCommit

**Comment by TheNeuralBit** [comment]
R: @reuvenlax could you take a look at this? It restores my changes to the schema proto, but with a modification to represent all logical types with a "javasdk" urn and serialized java object in the payload (for now).

I think the precommit failure is unrelated, Hannah tells me that https://github.com/apache/beam/pull/9610 should resolve it.

I ran the Dataflow VR tests and also merged in https://github.com/apache/beam/pull/9446 locally to make sure the `CloudObjectsTest` works

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SchemaTranslation.java [hunk 1]
```diff
@@ -19,44 +19,28 @@
...
-          .put(TypeName.MAP, RunnerApi.Schema.TypeName.MAP)
-          .put(TypeName.ROW, RunnerApi.Schema.TypeName.ROW)
-          .put(TypeName.LOGICAL_TYPE, RunnerApi.Schema.TypeName.LOGICAL_TYPE)
-          .build();
-
-  public static RunnerApi.Schema toProto(Schema schema) {
+
+  private static final String URN_BEAM_LOGICAL_DATETIME = "beam:fieldtype:datetime";
+  private static final String URN_BEAM_LOGICAL_DECIMAL = "beam:fieldtype:decimal";
+  private static final String URN_BEAM_LOGICAL_JAVASDK = "beam:fieldtype:javasdk";
```
Comment (robinyqiu) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SchemaTranslation.java]: Should we rename this URN to `beam:schema:logical_type:javasdk:v1` (similar above)? It sounds clearer to me, and I saw other URNs have version numbers
https://github.com/apache/beam/blob/909bc5a9cade71e484917d7627ab852928b44655/model/pipeline/src/main/proto/beam_runner_api.proto#L555

**Comment by Hannah-Jiang** [comment]
Run JavaPortabilityApi PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run Python PreCommit

**Comment by reuvenlax** [comment]
Run Dataflow ValidatesRunner

**Comment by reuvenlax** [comment]
Run SQL PostCommit

**Comment by TheNeuralBit** [comment]
Thank you @Hannah-Jiang! looks like syncing past #9610 fixed the precommit issue :)

**Comment by reuvenlax** [comment]
Run Dataflow Runner Nexmark Tests

**Comment by reuvenlax** [comment]
LGTM. Will merge when all tests pass.

**Comment by TheNeuralBit** [comment]
I'll write a patch for @robinyqiu's naming suggestions as well

**Comment by reuvenlax** [comment]
Run Dataflow Runner Nexmark Tests

**Comment by reuvenlax** [comment]
Run Dataflow ValidatesRunner

**Comment by reuvenlax** [comment]
Run SQL PostCommit

**Comment by reuvenlax** [comment]
Run Python PreCommit

**Comment by TheNeuralBit** [comment]
[BEAM-8286](https://issues.apache.org/jira/browse/BEAM-8286) seems to be the cause of the Python PreCommit failure

**Comment by TheNeuralBit** [comment]
Whoops fat fingered the "close and comment"

**Comment by TheNeuralBit** [comment]
Merged past #9620 which should fix the python precommit

**Comment by reuvenlax** [comment]
run Dataflow ValidatesRunner

**Comment by reuvenlax** [comment]
run Flink ValidatesRunner

**Comment by reuvenlax** [comment]
run SQL PostCommit

**Comment by reuvenlax** [comment]
run Portable Dataflow ValidatesRunner

**Pull request merged**